### PR TITLE
create form

### PR DESCRIPTION
**Pull request by vladib6**
Hi, i'd create a general form component

please check this out

### CODE REVIEW

File: evo-delivery-app/evo-delivery-frontend/src/hooks/use-form.ts [hunk 1]
```diff
@@ -0,0 +1,24 @@
+import { useState } from "react";
+import { FormFieldAttr } from "../pages/drivers/components/EvoForm";
+
+export const useForm = (callback: any, initialData={}) => {
```
Comment (tamirho) [inline evo-delivery-app/evo-delivery-frontend/src/hooks/use-form.ts:4]: `callback` 
maybe it's better to call it `handleSubmit`

File: evo-delivery-app/evo-delivery-frontend/src/index.css [hunk 2]
```diff
@@ -1,13 +1,67 @@
 body {
   margin: 0;
-  font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', 'Roboto', 'Oxygen',
-    'Ubuntu', 'Cantarell', 'Fira Sans', 'Droid Sans', 'Helvetica Neue',
+  font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", "Roboto", "Oxygen",
```
Comment (tamirho) [inline evo-delivery-app/evo-delivery-frontend/src/index.css:3]: please change your prettier configurations to use a single quote

File: evo-delivery-app/evo-delivery-frontend/src/pages/drivers/components/Driver/ViewDriver.tsx [hunk 3]
```diff
@@ -0,0 +1,64 @@
+<<<<<<< HEAD
```
Comment (tamirho) [inline evo-delivery-app/evo-delivery-frontend/src/pages/drivers/components/Driver/ViewDriver.tsx:1]: you need to fix that merge...

File: evo-delivery-app/evo-delivery-frontend/src/index.tsx [hunk 4]
```diff
@@ -45,7 +45,7 @@ root.render(
           <Route path={ENTITIES.drivers}>
             <Route index element={<Drivers />} />
             <Route path={`${ENTITY_VIEW_STATES.create}`} element={<CreateDriver />} />
-            <Route path={`:driverId/${ENTITY_VIEW_STATES.view}`} element={<Driver />} />
+            <Route path={`:driverId/${ENTITY_VIEW_STATES.view}`} element={<ViewDriver />} />
```
Comment (tamirho) [inline evo-delivery-app/evo-delivery-frontend/src/index.tsx:48]: Lets stick to the same naming conventions.
If you decide to change this component name, it's ok, but please change it for all entities..

File: evo-delivery-app/evo-delivery-frontend/src/pages/drivers/hooks/use-driver.tsx [hunk 5]
```diff
@@ -0,0 +1,7 @@
+import { useQuery } from "@tanstack/react-query";
+
+export const useDriver = (driverId: string) => {
+  return useQuery(["drivers", driverId], () =>
+    fetch(`api/v1/orders/${driverId}`).then((res) => res.json())
+  );
+};
```
Comment (tamirho) [inline evo-delivery-app/evo-delivery-frontend/src/pages/drivers/hooks/use-driver.tsx:7]: please use the `useGetEntity` hook

File: evo-delivery-app/evo-delivery-frontend/src/pages/drivers/hooks/use-drivers.tsx [hunk 6]
```diff
@@ -0,0 +1,7 @@
+import { useQuery } from "@tanstack/react-query";
+
+export const useDrivers = () => {
+  return useQuery(["drivers"], () =>
+    fetch("api/v1/drivers/").then((res) => res.json())
+  );
+};
```
Comment (tamirho) [inline evo-delivery-app/evo-delivery-frontend/src/pages/drivers/hooks/use-drivers.tsx:7]: please use the `useGetEntities` hook

File: evo-delivery-app/evo-delivery-frontend/src/pages/drivers/components/Drivers/Drivers.tsx [hunk 7]
```diff
@@ -1,5 +1,101 @@
...
+  const location = useLocation();
+  const goToEntity = useMemo(
+    () => (id: string) =>
+      navigate(`${id}/${ENTITY_VIEW_STATES.view}${location.search}`),
+    []
+  );
+
+  React.useEffect(() => {
+    return () => {};
+  }, []);
```
Comment (tamirho) [inline evo-delivery-app/evo-delivery-frontend/src/pages/drivers/components/Drivers/Drivers.tsx:47]: ?

File: evo-delivery-app/evo-delivery-frontend/src/pages/drivers/components/Drivers/Drivers.tsx [hunk 8]
```diff
@@ -1,5 +1,101 @@
...
+
+  return (
+    <EntityList
+      isLoading={false && (isFetching || isLoading)}
+      isError={false && isError}
```
Comment (tamirho) [inline evo-delivery-app/evo-delivery-frontend/src/pages/drivers/components/Drivers/Drivers.tsx:52]: remove `false &&` please

File: evo-delivery-app/evo-delivery-frontend/src/pages/drivers/components/Drivers/Drivers.tsx [hunk 9]
```diff
@@ -1,5 +1,101 @@
...
+      renderItem={(driver: Driver) => (
+        <>
+          <ListItem
+            key={driver._id}
+            disablePadding
+            alignItems="center"
+            secondaryAction={
+              <IconButton edge="end" aria-label="comments">
+                <DeleteIcon />
+              </IconButton>
```
Comment (tamirho) [inline evo-delivery-app/evo-delivery-frontend/src/pages/drivers/components/Drivers/Drivers.tsx:63]: please add `small` configurations as in [here](https://github.com/tamirho/evo-delivery/blob/e159b631cc9639968ba9d70e534fa288a234c494/evo-delivery-app/evo-delivery-frontend/src/pages/orders/components/Orders/Orders.tsx#L61-L63)

File: evo-delivery-app/evo-delivery-frontend/src/pages/drivers/components/Drivers/Drivers.tsx [hunk 10]
```diff
@@ -1,5 +1,101 @@
...
+            }
+          >
+            <ListItemButton onClick={() => goToEntity(driver._id)}>
+              <ListItemAvatar>
+                <Avatar>
+                  <InventoryIcon />
```
Comment (tamirho) [inline evo-delivery-app/evo-delivery-frontend/src/pages/drivers/components/Drivers/Drivers.tsx:69]: Try to find another icon for drivers ->  [`mui icons`](https://mui.com/material-ui/material-icons/)
This is used for orders

File: evo-delivery-app/evo-delivery-frontend/src/pages/drivers/components/Drivers/Drivers.tsx [hunk 11]
```diff
@@ -1,5 +1,101 @@
...
+                      color="text.primary"
+                    >
+                      {`${driver.name}`}
+                    </Typography>
+                    {` Capacity- ${driver.maxCapacity}`}
+                    {` Distance- ${driver.maxDistance}`}
+                  </>
+                }
+              />
+              <ListItemSecondaryAction></ListItemSecondaryAction>
```
Comment (tamirho) [inline evo-delivery-app/evo-delivery-frontend/src/pages/drivers/components/Drivers/Drivers.tsx:89]: Can remove that line

File: evo-delivery-app/evo-delivery-frontend/src/pages/drivers/components/EvoForm.tsx [hunk 12]
```diff
@@ -0,0 +1,54 @@
...
+    {}
+  );
+  const { onChange, onSubmit, values } = useForm(callback, data);
+
+  return (
+    <>
+      <form onSubmit={onSubmit}>
+        {initialData?.map((field) => (
+          <div>
+            <TextField
```
Comment (tamirho) [inline evo-delivery-app/evo-delivery-frontend/src/pages/drivers/components/EvoForm.tsx:33]: All our fields are text? What about number/int etc.?

File: evo-delivery-app/evo-delivery-frontend/src/pages/drivers/components/Drivers/Drivers.tsx [hunk 13]
```diff
@@ -1,5 +1,101 @@
...
+const drivers = [1, 2, 3, 4, 5, 6, 7, 8, 9, 2, 3, 4, 5, 6, 7, 8, 9].map(
+  (item, index) => ({
+    _id: `134523452345${item}${index}`,
+    name: `some name ${item}${index}`,
+    maxCapacity: index + 100,
+    maxDistance: index + 200,
+    createdAt: new Date(),
+    updatedAt: new Date(),
+  })
+);
```
Comment (tamirho) [inline evo-delivery-app/evo-delivery-frontend/src/pages/drivers/components/Drivers/Drivers.tsx:32]: try to integrate backend using the `useGetEntities` hook

**Review by tamirho — CHANGES_REQUESTED** [review]

File: evo-delivery-app/evo-delivery-frontend/src/pages/drivers/components/EvoForm.tsx (hunk 12 above)
Comment (vladib6) [reply evo-delivery-app/evo-delivery-frontend/src/pages/drivers/components/EvoForm.tsx:33]: it's just a wrapper to field , you can configure the type with "type" attribute , and i'm not sure we stay we this component maybe better to use <input> instead,,, we decide when we define the design

**Comment by vladib6** [comment]
all done

**Pull request merged**